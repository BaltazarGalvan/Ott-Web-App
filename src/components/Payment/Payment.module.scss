@use '../../styles/variables';
@use '../../styles/theme';
@use '../../styles/mixins/responsive';

.infoBox {
  display: flex;
  justify-content: space-between;
  margin-bottom: variables.$base-spacing;
  padding: variables.$base-spacing / 2 variables.$base-spacing;

  font-size: 14px;
  line-height: 18px;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);
  background: rgba(61, 59, 59, 0.08);
  border-radius: 4px;
  > strong {
    line-height: 16px;
    letter-spacing: 0.25px;
  }
}

.price {
  font-size: 14px;
  line-height: 18px;
  > strong {
    font-weight: bold;
    font-size: 24px;
    line-height: 26px;
  }
}

.cardDetails {
  display: flex;
}

.expiryDate {
  width: 250px;
}
