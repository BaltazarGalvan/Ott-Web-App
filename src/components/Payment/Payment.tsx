import React from 'react';
import { useTranslation } from 'react-i18next';
import type { Subscription } from 'types/subscription';

import Button from '../Button/Button';

import styles from './Payment.module.scss';

type Props = {
  subscription: Subscription;
  onEditSubscriptionClick: (subscription: Subscription) => void;
  panelClassName?: string;
  panelHeaderClassName?: string;
};

const Payment = ({ subscription, onEditSubscriptionClick, panelClassName, panelHeaderClassName }: Props): JSX.Element => {
  const { t } = useTranslation('user');
  const showAllTransactions = () => console.info('show all');

  return (
    <>
      <div className={panelClassName}>
        <div className={panelHeaderClassName}>
          <h3>{t('payment.subscription_details')}</h3>
        </div>
        <div className={styles.infoBox}>
          <p>
            <strong>{t('payment.monthly_subscription')}</strong> <br />
            {t('payment.next_billing_date_on')}
            {'<date>'}
          </p>
          <p className={styles.price}>
            <strong>{'â‚¬ 14.76'}</strong>
            {'/'}
            {t('payment.month')}
          </p>
        </div>
        <Button label={t('payment.edit_subscription')} onClick={() => onEditSubscriptionClick} />
      </div>
      <div className={panelClassName}>
        <div className={panelHeaderClassName}>
          <h3>{t('payment.payment_method')}</h3>
        </div>
        <div>
          <strong>{t('payment.card_number')}</strong>
          <p>xxxx xxxx xxxx 3456</p>
          <div className={styles.cardDetails}>
            <div className={styles.expiryDate}>
              <strong>{t('payment.expiry_date')}</strong>
              <p>{subscription.expiresAt}</p>
            </div>
            <div>
              <strong>{t('payment.cvc_cvv')}</strong>
              <p>******</p>
            </div>
          </div>
        </div>
      </div>
      <div className={panelClassName}>
        <div className={panelHeaderClassName}>
          <h3>{t('payment.transactions')}</h3>
        </div>
        <div className={styles.infoBox}>
          <p>
            <strong>{t('payment.monthly_subscription')}</strong> <br />
            {t('payment.price_payed_with_card')}
          </p>
          <p>
            {'<Invoice code>'}
            <br />
            {'<Date>'}
          </p>
        </div>
        <p>{t('payment.more_transactions', { amount: 4 })}</p>
        <Button label="Show all" onClick={() => showAllTransactions()} />
      </div>
    </>
  );
};

export default Payment;
