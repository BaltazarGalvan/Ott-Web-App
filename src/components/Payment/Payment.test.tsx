import React from 'react';
import { render } from '@testing-library/react';
import type { Subscription } from 'types/subscription';

import Payment from './Payment';

describe('<Payment>', () => {
  test('renders and matches snapshot', () => {
    const subscription = {} as Subscription;

    const { container } = render(<Payment subscription={subscription} update={(subscription) => console.info(subscription)} />);

    // todo
    expect(container).toMatchSnapshot();
  });
});
