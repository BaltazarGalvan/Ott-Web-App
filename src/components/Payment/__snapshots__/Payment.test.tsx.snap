// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Payment> renders and matches snapshot 1`] = `
<div>
  <div>
    <div>
      <h3>
        payment.subscription_details
      </h3>
    </div>
    <div
      class="infoBox"
    >
      <p>
        <strong>
          payment.monthly_subscription
        </strong>
         
        <br />
        payment.next_billing_date_on
        &lt;date&gt;
      </p>
      <p
        class="price"
      >
        <strong>
          â‚¬ 14.76
        </strong>
        /
        payment.month
      </p>
    </div>
    <button
      class="button default outlined"
    >
      <span>
        payment.edit_subscription
      </span>
    </button>
  </div>
  <div>
    <div>
      <h3>
        payment.payment_method
      </h3>
    </div>
    <div>
      <strong>
        payment.card_number
      </strong>
      <p>
        xxxx xxxx xxxx 3456
      </p>
      <div
        class="cardDetails"
      >
        <div
          class="expiryDate"
        >
          <strong>
            payment.expiry_date
          </strong>
          <p>
            03/2030
          </p>
        </div>
        <div>
          <strong>
            payment.cvc_cvv
          </strong>
          <p>
            ******
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <h3>
        payment.transactions
      </h3>
    </div>
    <div
      class="infoBox"
    >
      <p>
        <strong>
          payment.monthly_subscription
        </strong>
         
        <br />
        payment.price_payed_with_card
      </p>
      <p>
        &lt;Invoice code&gt;
        <br />
        &lt;Date&gt;
      </p>
    </div>
    <p>
      payment.more_transactions
    </p>
    <button
      class="button default outlined"
    >
      <span>
        Show all
      </span>
    </button>
  </div>
</div>
`;
